metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,592 @@
+    contact_point: str = Field(
+        default="localhost",
+        description="The cassandra instance contact point domain (without the port).",
+    )
+    port: str = Field(default="9042", description="The cassandra instance port.")
+    username: str = Field(
+        default="",
+        description=f"The username credential. Ensure user has read access to {SYSTEM_SCHEMA_KESPACE_NAME}.",
+    )
+    password: str = Field(default="", description="The password credential.")
+    keyspace_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="regex patterns for tables to filter in ingestion.",
+    )
+
+
+# source reporter
+@dataclass
+class CassandraSourceReport(SourceReport):
+    filtered: List[str] = field(default_factory=list)
+
+    def report_dropped(self, index: str) -> None:
+        self.filtered.append(index)
+
+
+# -------------------------------------------------- main source class -------------------------------------------------- #
+@platform_name("Cassandra")
+@config_class(CassandraSourceConfig)
+@support_status(SupportStatus.TESTING)
+@capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
+class CassandraSource(Source):
+
+    """
+    This plugin extracts the following:
+
+    - Metadata for tables
+    - Column types associated with each table column
+    - The keyspace each table belongs to
+    """
+
+    config: CassandraSourceConfig
+    report: CassandraSourceReport
+    cassandra_session: Session
+    platform: str
+
+    def __init__(self, ctx: PipelineContext, config: CassandraSourceConfig):
+        self.ctx = ctx
+        self.platform = PLATFORM_NAME_IN_DATAHUB
+        self.config = config
+        self.report = CassandraSourceReport()
+
+        # attempt to connect to cass
+        auth_provider: AuthProvider = None
+        if config.username and config.password:
+            ssl_context = SSLContext(PROTOCOL_TLSv1_2)
+            ssl_context.verify_mode = CERT_NONE
+            auth_provider = PlainTextAuthProvider(
+                username=config.username, password=config.password
+            )
+
+        cluster: Cluster = (
+            Cluster(
+                [config.contact_point],
+                port=config.port,
+                auth_provider=auth_provider,
+                ssl_context=ssl_context,
+            )
+            if auth_provider
+            else Cluster(
+                [config.contact_point],
+                port=config.port,
+            )
+        )
+        session: Session = cluster.connect()
+        self.cassandra_session = session
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = CassandraSourceConfig.parse_obj(config_dict)
+        return cls(ctx, config)
+
+    def get_workunits_internal(
+        self,
+    ) -> Iterable[MetadataWorkUnit]:
+
+        # get all keyspaces and iterate through them
+        # TODO: create a type for the response
```

Comment:
See for context https://github.com/datahub-project/datahub/pull/9680/files#r1460643089
Didn't impl it because we know the exact one column we want to get and this is unlikely to change. If there's strong preference to rigorously type everything and verify response against the types I'm happy to do it.